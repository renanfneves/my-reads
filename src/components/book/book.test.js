import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Book from './book';

const defaultProps = {
  id:"test-id",
  title:"mockedName",
  authors:["mockedAuthor"],
  imageLinks:{
    smallThumbnail: "mockedSmallThumbnail",
    thumbnail: "mockedThumbnail",
  },
  changeShelf: () => {},
};

it('renders book on none shelf', () => {
  const renderer = new ShallowRenderer();

  renderer.render(
    <Book
      {...defaultProps}
      shelf="none"
    />,
  );

  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});

it('renders book on currentlyReading shelf', () => {
  const renderer = new ShallowRenderer();

  renderer.render(
    <Book
      {...defaultProps}
      shelf="currentlyReading"
    />,
  );

  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});

it('renders book on wantToRead shelf', () => {
  const renderer = new ShallowRenderer();

  renderer.render(
    <Book
      {...defaultProps}
      shelf="wantToRead"
    />,
  );

  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});

it('renders book on read shelf', () => {
  const renderer = new ShallowRenderer();

  renderer.render(
    <Book
      {...defaultProps}
      shelf="read"
    />,
  );

  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
