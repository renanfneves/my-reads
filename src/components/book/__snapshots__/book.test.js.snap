// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders book on currentlyReading shelf 1`] = `
<div
  className="book"
>
  <div
    className="book-top"
  >
    <div
      className="book-cover"
      style={
        Object {
          "backgroundImage": "url(\\"mockedThumbnail\\")",
          "height": 193,
          "width": 128,
        }
      }
    />
    <div
      className="book-shelf-changer"
    >
      <ShelvesOptions
        changeShelf={[Function]}
        currentShelf="currentlyReading"
        id="test-id"
      />
    </div>
  </div>
  <div
    className="book-title"
  >
    mockedName
  </div>
  <div
    className="book-authors"
  >
    mockedAuthor
  </div>
</div>
`;

exports[`renders book on none shelf 1`] = `
<div
  className="book"
>
  <div
    className="book-top"
  >
    <div
      className="book-cover"
      style={
        Object {
          "backgroundImage": "url(\\"mockedThumbnail\\")",
          "height": 193,
          "width": 128,
        }
      }
    />
    <div
      className="book-shelf-changer"
    >
      <ShelvesOptions
        changeShelf={[Function]}
        currentShelf="none"
        id="test-id"
      />
    </div>
  </div>
  <div
    className="book-title"
  >
    mockedName
  </div>
  <div
    className="book-authors"
  >
    mockedAuthor
  </div>
</div>
`;

exports[`renders book on read shelf 1`] = `
<div
  className="book"
>
  <div
    className="book-top"
  >
    <div
      className="book-cover"
      style={
        Object {
          "backgroundImage": "url(\\"mockedThumbnail\\")",
          "height": 193,
          "width": 128,
        }
      }
    />
    <div
      className="book-shelf-changer"
    >
      <ShelvesOptions
        changeShelf={[Function]}
        currentShelf="read"
        id="test-id"
      />
    </div>
  </div>
  <div
    className="book-title"
  >
    mockedName
  </div>
  <div
    className="book-authors"
  >
    mockedAuthor
  </div>
</div>
`;

exports[`renders book on wantToRead shelf 1`] = `
<div
  className="book"
>
  <div
    className="book-top"
  >
    <div
      className="book-cover"
      style={
        Object {
          "backgroundImage": "url(\\"mockedThumbnail\\")",
          "height": 193,
          "width": 128,
        }
      }
    />
    <div
      className="book-shelf-changer"
    >
      <ShelvesOptions
        changeShelf={[Function]}
        currentShelf="wantToRead"
        id="test-id"
      />
    </div>
  </div>
  <div
    className="book-title"
  >
    mockedName
  </div>
  <div
    className="book-authors"
  >
    mockedAuthor
  </div>
</div>
`;
